ToDoApp/Controllers/MainViewController.swift

```diff
@@ -7,128 +7,198 @@
+    cell.titleLabel.text = items[indexPath.row].title
+    cell.deadlineLabel.text = items[indexPath.row].date
+    cell.priorityLabel.text = items[indexPath.row].priority
+    
+    configureCompletedImageView(cell, items[indexPath.row].isCompleted)
+    if let priorityOption = PriorityOptions(rawValue: cell.priorityLabel.text ?? "") {
+        configurePriorityLabel(cell, priorityOption)
     }
+    cell.selectionStyle = .none
+    cell.makeUI()
+    
+    return cell
+  }
+  
+  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    tableView.deselectRow(at: indexPath, animated: true)
     
-    // MARK: - cell에 데이터 전달
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: CustomCell.cellId, for: indexPath) as! CustomCell
-        
-        cell.titleLabel.text = items[indexPath.row].title
-        cell.deadlineLabel.text = items[indexPath.row].date
-        cell.priorityLabel.text = items[indexPath.row].priority
-        
-        configureCompletedImageView(cell, items[indexPath.row].isCompleted)
-        configurePriorityLabel(cell, cell.priorityLabel.text!)
-        
-        return cell
+    let selectedItem = items[indexPath.row]
+    let editItemViewController = AddItemViewController()
+    editItemViewController.itemToEdit = selectedItem
+    self.navigationController?.pushViewController(editItemViewController, animated: true)
+    editItemViewController.delegate = self
+    itemToEdit = selectedItem
+  }
+ 
+  func tableView(_ tableView: UITableView,
+                 trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath)
+                  -> UISwipeActionsConfiguration? {
+      let deleteAction = UIContextualAction(style: .destructive, title: "삭제") { (action, view, completion) in
+          self.items.remove(at: indexPath.row)
+          tableView.deleteRows(at: [indexPath], with: .fade)
+          completion(true)
+      }
+      let swipeConfiguration = UISwipeActionsConfiguration(actions: [deleteAction])
+      swipeConfiguration.performsFirstActionWithFullSwipe = false
+      
+      return swipeConfiguration
+  }
+  // MARK: - completed 확인함수
+  func configureCompletedImageView(_ cell: CustomCell, _ isCompleted: Bool) {
+    if isCompleted {
+      cell.completedImageView.image = UIImage(systemName: "checkmark.circle")
+      cell.completedImageView.tintColor = UIColor.systemGreen
+    } else {
+      cell.completedImageView.image = nil
     }
+  }
+  
+  // MARK: - 우선순위확인함수, enum사용해서
+  func configurePriorityLabel(_ cell: CustomCell, _ priority: PriorityOptions) {
+      switch priority {
+      case .low:
+          cell.priorityLabel.backgroundColor = Color.low.uiColor
+      case .medium:
+          cell.priorityLabel.backgroundColor = Color.medium.uiColor
+      case .high:
+          cell.priorityLabel.backgroundColor = Color.high.uiColor
+      }
```

Comment:
`enum`의 값에 따라 다른 값을 변경해야하는 경우라면 
`Color.swift` 파일에서 한 것 처럼 `enum`에 computed property를 사용해주면 좋을 듯!

```suggestion
      cell.priorityLabel.backgroundColor = priority.tintColor
```